modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/util/cropregion/get_crop_region.es.js

```diff
@@ -0,0 +1,59 @@
+import {isObject} from 'metal';
+
+/**
+ * Returns dimensions and coordinates representing a cropped region
+ * @param {HTMLElement} imagePreview Image
+ * @param {!region} region Object
+ * @return {object} Object representing dimensions and coordinates of a cropped region
+ * @review
+ */
+
+export default function getCropRegion(imagePreview, region) {
+	if (imagePreview.tagName === 'IMG' && isObject(region)) {
+		let naturalSize = _getImageNaturalSize(imagePreview);
+
+		let scaleX = naturalSize.width / imagePreview.offsetWidth;
+		let scaleY = naturalSize.height / imagePreview.offsetHeight;
+
+		let regionHeight = region.height ? (region.height * scaleY) : naturalSize.height;
+		let regionWidth = region.width ? (region.width * scaleX) : naturalSize.width;
+
+		let regionX = region.x ? Math.max(region.x * scaleX, 0) : 0;
+		let regionY = region.y ? Math.max(region.y * scaleY, 0) : 0;
+
+		return {
+			height: regionHeight,
+			width: regionWidth,
+			x: regionX,
+			y: regionY
+		};
+	}
+
+	return null;
+}
+
+/**
+ * Return image's natural size or create a new Image element if it doesn't exist
+ * @param {HTMLElement} image
+ * @return {Object} Return object containing natural height and width of a received or created image
+ * @private
+ * @review
+ */
+function _getImageNaturalSize(image) {
```

Comment:
We don't need this function, it's only purpose is compatibility with IE older than 9.0. We can assume all images have 'natural' properties.